android-client-sdk/src/main/java/com/devcycle/sdk/android/model/User.kt

```diff
@@ -76,15 +76,15 @@ internal data class User constructor(
             return User(userId, isAnonymous)
         }
 
-        @JvmSynthetic internal fun fromUserParam(user: DVCUser, context: Context): User {
+        @JvmSynthetic internal fun fromUserParam(user: DVCUser, context: Context, anonUserId: String?): User {
             val locale: Locale = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
                 context.resources.configuration.locales[0]
             } else {
                 context.resources.configuration.locale
             }
 
             val isAnonymous = user.isAnonymous ?: false
-            val userId = if (isAnonymous) UUID.randomUUID().toString() else user.userId
+            val userId = if (isAnonymous) if(anonUserId == null) UUID.randomUUID().toString() else anonUserId else user.userId
```

Comment:
the nested ternary on a single line doesn't make this very readable, maybe we could split it up?
```suggestion
            val userId = if (isAnonymous) {
                if (anonUserId == null) UUID.randomUUID().toString() else anonUserId
            } else {
                user.userId
            }
```